### PR TITLE
Feature/get all graph items

### PR DESCRIPTION
**Pull request by shanefontaine**
Apparently TheGraph only allows for `skip`ping up to 6000 items.

* [Discord conversations](https://discord.com/channels/438038660412342282/438070183794573313/862337114019463199)
* Error message when we run large queries: `The skip argument must be between 0 and 5000, but is 6000`

The [docs](https://thegraph.com/docs/en/developer/graphql-api#pagination) mention "For retrieving a large number of items, it is much better to page through entities based on an attribute as shown in the last example.". Chris' recent TheGraph queries work this way as well. This PR attempts to do this by returning all items and using the ID as the index for the query and returning all IDs instead of being capped at 6000.

Only important note is that the order always needs to be `asc` for these searches, since it is "counting" up from `0x00000...` to `0xfffff` with this method.

### CODE REVIEW

File: packages/hop-node/src/theGraph/getBondedWithdrawals.ts [hunk 1]
```diff
@@ -26,7 +31,20 @@ export default async function getBondedWithdrawals (chain: string, token: string
...
   `
   const jsonRes = await makeRequest(chain, query, {
-    token
+    token,
+    lastId
   })
-  return jsonRes.withdrawalBondeds.map((x: any) => normalizeEntity(x))
+  let withdrawals = jsonRes.withdrawalBondeds.map((x: any) => normalizeEntity(x))
+
+  lastId = withdrawals[withdrawals.length - 1].id
```
Comment (miguelmota) [inline packages/hop-node/src/theGraph/getBondedWithdrawals.ts:39]: is it possible that `withdrawals.length` here can be `0` so `.id` would throw an error trying to read from `undefined`?

File: packages/hop-node/src/theGraph/getTransferIds.ts [hunk 2]
```diff
@@ -46,34 +46,26 @@ export default async function getTransferIds (
...
-        token,
-        filters,
-        skip + 1000
-      ))
-    } catch (err) {
-      if (!err.message.includes('The `skip` argument must be between')) {
-        throw err
-      }
-    }
+  lastId = transfers[transfers.length - 1].id
```
Comment (miguelmota) [inline packages/hop-node/src/theGraph/getTransferIds.ts:60]: is it possible that `transfers.length` here can be `0` so `.id` would throw an error trying to read from `undefined`?

File: packages/hop-node/src/theGraph/getTransferRoots.ts [hunk 3]
```diff
@@ -30,23 +34,19 @@ export default async function getTransferRoots (chain: string, token: string, sk
...
-        chain,
-        token,
-        skip + 1000
-      ))
-    } catch (err) {
-      if (!err.message.includes('The `skip` argument must be between')) {
-        throw err
-      }
-    }
+  lastId = roots[roots.length - 1].id
```
Comment (miguelmota) [inline packages/hop-node/src/theGraph/getTransferRoots.ts:42]: is it possible that `roots.length` here can be `0` so `.id` would throw an error trying to read from `undefined`?

**Review by miguelmota — COMMENTED** [review]
Left minor code comments. I think the changes make sense. I'll try to run it as well to make sure it works as expected

File: packages/hop-node/src/theGraph/getBondedWithdrawals.ts (hunk 1 above)
Comment (shanefontaine) [reply packages/hop-node/src/theGraph/getBondedWithdrawals.ts]: Yes, great call. Fixed with 26ffbf5f93e5a0d1cd10ed6b2e87d8b1919e4177

File: packages/hop-node/src/theGraph/getTransferIds.ts (hunk 2 above)
Comment (shanefontaine) [reply packages/hop-node/src/theGraph/getTransferIds.ts]: Yes, great call. Fixed with 26ffbf5f93e5a0d1cd10ed6b2e87d8b1919e4177

File: packages/hop-node/src/theGraph/getTransferRoots.ts (hunk 3 above)
Comment (shanefontaine) [reply packages/hop-node/src/theGraph/getTransferRoots.ts]: Yes, great call. Fixed with 26ffbf5f93e5a0d1cd10ed6b2e87d8b1919e4177

**Comment by shanefontaine** [comment]
Comments addressed. Back to you.

@miguelmota

**Comment by miguelmota** [comment]
Changes look great and works well 👍

**Pull request merged**